include/boost/url/impl/segments.hpp

```diff
@@ -29,19 +31,25 @@ namespace urls {
 
 class segments::iterator
 {
-    url const* u_ = nullptr;
-    std::size_t i_ = 0;
-    const_string::factory a_;
+    detail::segments_iterator_base impl_;
```

Comment:
if you are going to use composition then don't call them bases! e.g. `segments_iter_impl`